### PR TITLE
[Kotlin Migration] BasicModel

### PR DESCRIPTION
**Pull request by n1snt**
## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented on your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: api/src/main/java/com/ichi2/anki/api/BasicModel.kt [hunk 1]
```diff
@@ -1,14 +1,18 @@
 //noinspection MissingCopyrightHeader #8659
-package com.ichi2.anki.api;
+package com.ichi2.anki.api
 
 /**
  * Definitions of the basic model
  */
-class BasicModel {
-    public static final String[] FIELDS = {"Front","Back"};
+internal object BasicModel {
```
Comment (david-allison) [inline api/src/main/java/com/ichi2/anki/api/BasicModel.kt:7]: Why is this now internal?

**Comment by david-allison** [comment]
@mikehardy I feel this warrants a bump in the major version of the API project (Kotlin visibility change on package protected). Do you agree?

It's a breaking change which probably won't affect consumers as they're unlikely to put code under our package.

**Comment by mikehardy** [comment]
I'll trust you on that - I haven't reviewed and am heading away from keyboard from a while
but philosophically I'm a semver robot: if it smells like a breaking change it probably is, version numbers are free, documentation explaining the break is golden, so add a changelog doc, do the major version and move on. zero drama, it's unemotional

**Comment by david-allison** [comment]
Let's wait to confirm then, no need to rush this

**Comment by n1snt** [comment]
Yeah, I too think this justifies a bump in the major version of the API project.
I will update it to use `internal` 👍

File: api/src/main/java/com/ichi2/anki/api/BasicModel.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+//noinspection MissingCopyrightHeader #8659
+package com.ichi2.anki.api
+
+/**
+ * Definitions of the basic model
+ */
+internal class BasicModel {
```
[the remark this answers is not in the document]
Comment (mikehardy) [reply api/src/main/java/com/ichi2/anki/api/BasicModel.kt:7]: > Should I create another issue "release a new version of API"? 

Yep, you can use the three-dot menu at top right of the comment to create a new issue and it pre-populates it with text of my comment and helpful links back to the comment that generated the issue

> Sorry I didn't understand what you mean by "putting a v2.0.0 here".

I mean literally change that string to 2.0.0, that's how you change the version

[the remark this answers is not in the document]
Comment (n1snt) [reply api/src/main/java/com/ichi2/anki/api/BasicModel.kt:7]: Created the issue.
Should I change the comment to "2.0.0"? or the version string in build.gradle file?

[the remark this answers is not in the document]
Comment (mikehardy) [reply api/src/main/java/com/ichi2/anki/api/BasicModel.kt:7]: Go ahead and change the version to 2 in this PR so we don't forget, and this is the PR that creates the break, so having the version changed at same time makes sense. The API module is not auto-versioned like the AnkiDroid module

**Review by david-allison — CHANGES_REQUESTED** [review]
Commit 1: Explain in the commit message why we're bumping the version number

**Comment by n1snt** [comment]
Done

**Review by david-allison — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
nice!

**Pull request merged**